theano/compile/builders.py

```diff
@@ -101,14 +102,6 @@ def __init__(self, inputs, outputs, **kwargs):
         self.input_types = [input.type for input in inputs]
         self.output_types = [output.type for output in outputs]
 
-    def __eq__(self, other):
-        # TODO: recognize a copy
-        return self is other
```

Comment:
Keep \__eq__, it don't do the same as the default. And keep the comment. Also keep \__hash__ and its todo. remove \__props__ for this op, it is a special op.